### PR TITLE
Rework ID-based parametrisation

### LINKED ISSUE #24: Adding a var causes incorrect output
I think there is a chance I'm hitting this "Currently, adding one more variable here causes the model to incorrectly load the static variables. It is possible to hack around this. We are working on a more principled fix. Please open an issue if this is a problem for you." So I'm logging an issue to identify the workaround.

Repro steps:
1. Ensure ERA5 data has additional variables beyond the default (I randomly picked the tsn var for this example) & that batch is loading that var. Also precalculate the mean and std for the new var
```
from aurora import normalisation
#extend locations to include the mean of tsn
normalisation.locations["tsn"] = np.float32(surf_vars_ds.tsn.mean().values)
normalisation.scales["tsn"] = np.float32(surf_vars_ds.tsn.std().values)
```

2. Load the pretrained weights
```
pretrained_weights = torch.load("aurora-0.25-small-pretrained.ckpt")
```
3. Adjust the pretrained weights to accommodate the new var & initialize to pretrained weights
```
new_surf_token_embeds_weight = torch.zeros((256, 8, 2, 4, 4))  
new_surf_token_embeds_weight[:, :7, :, :, :] = pretrained_weights['net.encoder.surf_token_embeds.weight']
pretrained_weights['net.encoder.surf_token_embeds.weight'] = new_surf_token_embeds_weight

new_surf_head_weight = torch.zeros((80, 512))  
new_surf_head_weight[:64, :] = pretrained_weights['net.decoder.surf_head.weight']
pretrained_weights['net.decoder.surf_head.weight'] = new_surf_head_weight

new_surf_head_bias = torch.zeros((80,))  
new_surf_head_bias[:64] = pretrained_weights['net.decoder.surf_head.bias']
pretrained_weights['net.decoder.surf_head.bias'] = new_surf_head_bias
```
4. Fixup the names (i have no idea why these come downloaded with "net" prefix which I'm just removing)
```
new_pretrained_weights = {}
for key, value in pretrained_weights.items():
    new_pretrained_weights[key[4:]] = value
```
5. Initialize the model with the extra var
```
from aurora import AuroraSmall

model = AuroraSmall(surf_vars=("2t", "10u", "10v", "msl", "tsn"), use_lora=False)

model.load_state_dict(new_pretrained_weights)
```
6. Make a prediction 
```
from aurora import Aurora, rollout

model.eval()
model = model.to("cuda")

with torch.inference_mode():
    preds = [pred.to("cpu") for pred in rollout(model, batch, steps=2)]

model = model.to("cpu")
```
7. Plot the output of the 2t which was in the original model

image

Expected result (plot of 2t using default input vars)

image

Its very possible I missed an assumption as part of this process but given the comment referenced above about adding the var and how the output looks I figured there was a chance I was running in to the referenced issue.

Also note that I subsetted the batch input to a smaller input set.  It seemed that this was mostly handled in the model initialization but wanted to note that in case it was related.
Thanks!

### PR DESCRIPTION
**Pull request by wesselb**
This should make it easier to extend the model to new variables.

- [x] Check correctness in notebooks. Looks good.
- [x] Remove warnings.
- [x] Add example in docs.

Closes #24

### CODE REVIEW

File: docs/finetuning.md [hunk 1]
```diff
@@ -10,21 +10,49 @@ model = Aurora(use_lora=False)  # Model is not fine-tuned.
...
+# Normalisation standard deviations:
+scales["new_surf_var"] = 1.0
+scales["new_static_var"] = 1.0
+scales["new_atmos_var"] = 1.0
+```
+
+## Other Model Extensions
+
+It is possible to extend to model in any way you like.
+If you do this, you will likely you add or remove parameters.
```
Comment (a-lucic) [inline docs/finetuning.md:45]:
```suggestion
If you do this, you will likely add or remove parameters.
```

**Review by a-lucic — APPROVED** [review]
LGTM. Happy this is implemented!

**Comment by scottcha** [comment]
I checked this PR against my fine tuning and was able to add variables w/out needing to directly modify the layers (all completed on the Small version of Aurora).  Adding the ERA5 tsn var w/out any additional fine tuning results in 
image

After a few mins of fine tuning I get this for tsn
image

For the existing 2t var I get this w/out finetuning
image

And this after a few mins of fine tuning
image

While I'm not super experienced yet with leveraging this model so far the behavior seems to be expected and its possible to add vars more easily than previously.

**Comment by wesselb** [comment]
Thanks, @scottcha! I'm going to interpret this as evidence that the refactoring works as intended and proceed to merging. :)

**Pull request merged**